### PR TITLE
SYNR-1128

### PR DESCRIPTION
**Pull request by kimyen**

### CODE REVIEW

**Comment by kimyen** [comment]
@brucehoff I am looking into the tests. What do you think about this approach?

**Comment by kimyen** [comment]
I am working on (1) and taking notes for (2). I think there is an open issue for (2) SYNR-1011, but I can also checkin the table vignettes work here so that we can review them altogether.

**Comment by brucehoff** [comment]
OK.  The approach looks good, just want to see what it looks like in
practice.
Thanks,
Bruce

**Comment by kimyen** [comment]
added the tests
``` R CMD check . ``` output:
``` 
* checking tests ...
  Running ‘testthat.R’
 OK
```
Will setup my private build for PythonEmbedInR

File: R/PyGet.R [hunk 1]
```diff
@@ -183,7 +183,7 @@ setMethod("pyGetPoly", signature(key="character", autoTypecast = "logical", simp
 setClass("DataFrame")
 setMethod("pyGetPoly", signature(key="character", autoTypecast = "logical", simplify = "logical", pyClass = "DataFrame"),
           function(key, autoTypecast, simplify, pyClass){
-    x <- pyExecg(sprintf("x = %s.to_dict()", key), autoTypecast = autoTypecast, simplify = simplify)[["x"]]
-    return( as.data.frame(x, stringsAsFactors=FALSE) )
+    pyExec(sprintf("x = %s.to_dict(orient='list')", key))
```
Comment (kimyen) [inline R/PyGet.R]: Make notes in vignettes to ask for user feedback

File: tests/testthat/install_pandas.R [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# install pandas 
+
+# first import pandas in python
+baseDir <- getwd()
+pyExec(sprintf("sys.path.append(\"%s\")", file.path(baseDir, "tests", "testthat")))
+pyImport("install_pandas")
+pyExec(sprintf("install_pandas.main('%s')", baseDir))
```
Comment (kimyen) [inline tests/testthat/install_pandas.R]: could `R CMD check` run at a different dir other than PythonEmbedInR?

File: tests/testthat/test_pyCall.R [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  expect_equal(class(df), "data.frame")
+  
+  pySet("df", df)
+  expect_output(pyExecp("type(df)"), "pandas.core.frame.DataFrame")
+
+  # define a test function in python
+  pyExec("return_df = lambda: df")
+
+  df2 <- pyCall("return_df")
+  expect_equal(class(df2), "data.frame")
```
Comment (kimyen) [inline tests/testthat/test_pyCall.R]: should clean up python pandas after the test (or install on a different location and ignore this location)

**Comment by kimyen** [comment]
`R CMD check` writes to PythonEmbedInR/..Rcheck directory. While executing the tests as part of the check command, `getwd()` returns `.../PythonEmbedInR/..Rcheck/tests`.

**Comment by kimyen** [comment]
`helper.R` runs before all tests. However, `teardown-prefix` is still an open issue:https://github.com/hadley/testthat/issues/589

File: tests/testthat/helper_pandas.R [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# helper that load before tests are run
+
+# determine the package directory path
+package_dir <- gsub("(PythonEmbedInR).*", "\\1", getwd())
```
Comment (brucehoff) [inline tests/testthat/helper_pandas.R]: for those of us weak in regexp's, please add a comment explaining what this does, as well as a couple tests that exercise it.

File: tests/testthat.R [hunk 5]
```diff
@@ -0,0 +1,9 @@
+library(testthat)
+library(PythonEmbedInR)
+
+test_check("PythonEmbedInR")
+
+# uninstall pandas
+print("Uninstalling pandas ...")
+package_dir <- gsub("(PythonEmbedInR).*", "\\1", getwd())
```
Comment (brucehoff) [inline tests/testthat.R]: for those of us weak in regexp's, please add a comment explaining what this does, as well as a couple tests that exercise it.

**Pull request merged**